### PR TITLE
[BG-6488] Port signing feature to admin tool with v2 coins

### PR DESCRIPTION
**Pull request by dacruz21**
Moves the signing logic from `signoffline.js` to `admin.js sign`.

- [x] UTXO recoveries
- [x] Segwit inputs for UTXO recoveries
- [x] Bech32 inputs for UTXO recoveries
- [x] Ethereum recoveries 
- [x] XRP recoveries
- [x] XLM recoveries
- [x]  ERC20 recoveries

### CODE REVIEW

File: app/sign.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  btc: utxoLib.networks.bitcoin,
+  ltc: utxoLib.networks.litecoin,
+  bch: utxoLib.networks.bitcoincash,
+  zec: utxoLib.networks.zcash,
+  tltc: utxoLib.networks.litecoin,
+  tbtc: utxoLib.networks.testnet
+};
+
+const coinDecimals = {
+  btc: 8,
```
Comment (marktoda) [inline app/sign.js]: Maybe make all of these BN too so ETH isn't such a lonely edge case?

File: app/admin.js [hunk 2]
```diff
@@ -34,6 +35,30 @@ importKeys.addArgument(
...
+signCommand.addArgument(
+  ['key'],
+  {
+    action: 'store',
+    required: false, // can be typed during the signing process to avoid leaving the xprv in the shell history
+    help: 'private key to sign the transaction with'
+  }
+);
+signCommand.addArgument(
+  ['--confirm'],
```
Comment (marktoda) [inline app/admin.js]: Is this just for automated signing? Should we allow automated signing?

File: app/sign.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+  if (backupKeyNode.toBase58() === backupKeyNode.neutered().toBase58()) {
+    throw new Error('please provide the private (not public) wallet key');
+  }
+
+  if (backupKeyNode.neutered().toBase58() !== recoveryRequest.backupKey) {
+    throw new Error('provided private key does not match public key specified with recovery request');
+  }
+
+  if (!skipConfirm) {
```
Comment (marktoda) [inline app/sign.js]: Throw this confirmy branch into a separate function, it's duped in a couple other places

File: app/sign.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+};
+
+const handleSignXrp = function(recoveryRequest, key, skipConfirm) {
+  const transaction = rippleParse.decode(recoveryRequest.txHex);
+  const rippleApi = new rippleLib.RippleAPI();
+  const outputAddress = transaction.Destination;
+  const outputAmount = transaction.Amount / Math.pow(10, coinDecimals.xrp);
+  const customMessage = recoveryRequest.custom ? recoveryRequest.custom.message : "None";
+
+  console.log('Sign Recovery Transaction');
```
Comment (marktoda) [inline app/sign.js]: Throw this messaging into a function, code duplication

File: app/admin.js (hunk 2 above)
Comment (dacruz21) [reply app/admin.js]: We allowed it in version 1 per Keyternal's request. It's also useful for unit tests.

File: app/sign.js [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+  }
+
+  if (backupKeyNode.neutered().toBase58() !== recoveryRequest.backupKey) {
+    throw new Error('provided private key does not match public key specified with recovery request');
+  }
+
+  const backupHDNode = prova.HDNode.fromBase58(key);
+  const backupSigningKey = backupHDNode.getKey().getPrivateKeyBuffer();
+
+  transaction.sign(backupSigningKey);
```
Comment (marktoda) [inline app/sign.js]: Are we just leaving it up to the KRS to build the ETH function call to the contract and stuff before signing? Separately, as in BGJS/eth.js:264-5, they create the operationHash specifically with the contract function call data and then signs that. And it looks like ethereumjs-tx computes the hash differently

Comment (dacruz21) [reply app/sign.js]: KRS doesn't build the function call - the user does through the WRW, or BitGo does through bgadmin.

**Pull request merged**